### PR TITLE
Change default on delete to restrict for required relations

### PR DESCRIPTION
**Pull request by tomhoule**

### CODE REVIEW

**Comment by pimeys** [comment]
```

[2020-12-02T10:19:20Z] Dec 02 10:19:20.459 ERROR tiberius::tds::stream::token: Incorrect syntax near the keyword 'RESTRICT'. code=156
--


```

**Comment by pimeys** [comment]
```
  | [ FOREIGN KEY ]
        REFERENCES [ schema_name . ] referenced_table_name [ ( ref_column ) ]
        [ ON DELETE { NO ACTION | CASCADE | SET NULL | SET DEFAULT } ]
        [ ON UPDATE { NO ACTION | CASCADE | SET NULL | SET DEFAULT } ]
        [ NOT FOR REPLICATION ]
```
https://docs.microsoft.com/en-us/sql/t-sql/statements/create-table-transact-sql?view=sql-server-ver15

File: libs/prisma-models/src/datamodel_converter.rs [hunk 1]
```diff
@@ -210,24 +210,24 @@ impl<'a> DatamodelConverter<'a> {
...
-                                panic!("It's not allowed that both sides of a relation specify the inline policy. The field was {} on model {}. The related field was {} on model {}.", field.name, model.name, related_field.name, related_model.name)
-                            }
-                        },
-                    };
+                        }
+                        (Some(_), Some(_)) => {
+                            panic!("It's not allowed that both sides of a relation specify the inline policy. The field was {} on model {}. The related field was {} on model {}.", field.name, model.name, related_field.name, related_model.name)
+                        }
+                    },
+                };
```
Comment (mavilein) [inline libs/prisma-models/src/datamodel_converter.rs:230]: this is just `cargo fmt`

File: migration-engine/migration-engine-tests/tests/migration_tests.rs [hunk 2]
```diff
@@ -1968,10 +1968,10 @@ async fn join_tables_between_models_with_compound_primary_keys_must_work(api: &T
             .assert_has_column("cat_id")?
             .assert_fk_on_columns(&["human_firstName", "human_lastName"], |fk| {
                 fk.assert_references("Human", &["firstName", "lastName"])?
-                    .assert_cascades_on_delete()
+                    .assert_restrict_on_delete()
```
Comment (mavilein) [inline migration-engine/migration-engine-tests/tests/migration_tests.rs:1971]: I am not sure why this doesn't fail for SQL Server as we are using `NO ACTION` for this database.

Comment (mavilein) [reply migration-engine/migration-engine-tests/tests/migration_tests.rs]: It did fail. I just did not push the latest version :-D

**Comment by tomhoule** [comment]
@mavilein should we close this?

**Comment by tomhoule** [comment]
This is going to be revisited soon. Proposal: https://github.com/prisma/prisma/issues/6996

**Pull request closed without merging**